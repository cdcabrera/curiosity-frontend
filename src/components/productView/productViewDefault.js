import React from 'react';
import { ProductView } from './productView';

/**
 * A default product view.
 *
 * @returns {Node}
 */
const ProductViewDefault = () => <ProductView />;

/**
 * Prop types.
 */
ProductViewDefault.propTypes = {};

/**
 * Default props.
 */
ProductViewDefault.defaultProps = {};

export { ProductViewDefault as default, ProductViewDefault };
