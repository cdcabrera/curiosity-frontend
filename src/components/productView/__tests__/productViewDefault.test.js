import React from 'react';
import { shallow } from 'enzyme';
import { ProductViewDefault } from '../productViewDefault';

describe('ProductViewDefault Component', () => {
  it('should render a non-connected component', () => {
    const props = {
      routeDetail: {
        pathParameter: 'lorem ipsum',
        productConfig: [{ lorem: 'ipsum' }],
        productParameter: 'lorem ipsum product label'
      }
    };

    const component = shallow(<ProductViewDefault {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });
});
