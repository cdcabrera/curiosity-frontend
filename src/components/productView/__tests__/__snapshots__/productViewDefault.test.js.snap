// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductViewDefault Component should render a non-connected component: non-connected 1`] = `
<ProductView
  routeDetail={
    Object {
      "pathParameter": "lorem ipsum",
      "productConfig": Array [
        Object {
          "lorem": "ipsum",
        },
      ],
      "productParameter": "dolor sit",
    }
  }
  t={[Function]}
  toolbarGraph={null}
  toolbarGraphDescription={false}
  toolbarProduct={null}
/>
`;
